services/ui-src/src/components/pages/Dashboard/DashboardPage.test.tsx

```diff
@@ -47,29 +54,161 @@ const dashboardViewWithReports = (
   </RouterWrappedComponent>
 );
 
-describe("Test Report Dashboard view (with reports, desktop view)", () => {
-  beforeEach(() => {
+describe("Test Report Dashboard view (Desktop)", () => {
+  beforeEach(async () => {
     mockedUseUser.mockReturnValue(mockStateUser);
     mockUseBreakpoint.mockReturnValue({
       isMobile: false,
     });
     mockMakeMediaQueryClasses.mockReturnValue("desktop");
-    mockedUseStore.mockReturnValue(mockReportStore);
   });
 
   afterEach(() => {
     jest.clearAllMocks();
   });
 
-  test("Check that WP Dashboard view renders", async () => {
-    await act(async () => {
-      await render(dashboardViewWithReports);
+  describe("Desktop view (basic)", () => {
+    test("Check that WP Dashboard view renders", () => {
+      mockedUseStore.mockReturnValue(mockReportStore);
+      render(dashboardViewWithReports);
+
+      expect(screen.getByText(wpVerbiage.intro.header)).toBeVisible();
+      expect(
+        screen.queryByText(wpVerbiage.body.table.caption)
+      ).toBeInTheDocument();
+      expect(screen.queryByText(wpVerbiage.body.empty)).not.toBeInTheDocument();
+      expect(screen.queryByText("Leave form")).not.toBeInTheDocument();
+    });
+  });
+
+  describe("Desktop view with report, state & banner ", () => {
+    beforeEach(() => {
+      mockedUseStore.mockReturnValue(mockUseStore);
+      render(dashboardViewWithReports);
+    });
+
+    test("Clicking Call To Action text open add/edit modal", async () => {
+      const callToActionButton = screen.getByText(wpVerbiage.body.callToAction);
+      expect(callToActionButton).toBeVisible();
+      await userEvent.click(callToActionButton);
+      expect(screen.queryByText("Start new")).toBeVisible();
+    });
+  });
+});
+
+describe("Test Report Dashboard (Mobile)", () => {
+  beforeEach(() => {
+    mockUseBreakpoint.mockReturnValue({
+      isMobile: true,
+    });
+
+    mockedUseStore.mockReturnValue(mockUseStore);
+    render(dashboardViewWithReports);
+  });
+
+  test("Clicking Call To Action text open add/edit modal", async () => {
+    const callToActionButton = screen.getByText(wpVerbiage.body.callToAction);
+    expect(callToActionButton).toBeVisible();
+    await userEvent.click(callToActionButton);
+    expect(screen.queryByText("Start new")).toBeVisible();
+  });
+});
```

Comment:
same question

Reply:
same as above

Reply:
keeping this one as is because i do plan to add 1 more test case here once sars is in and working